Automattic/jetpack
# Mailchimp block: Add Align, Color and Spacing features

The proposed changes add Align (Wide & Full Width), Color and Spacing (Padding & Margin) features to the Mailchimp block:

 Before | After |
| ------------- | ------------- |
| ![Markup on 2022-02-03 at 11:18:11](https://user-images.githubusercontent.com/25105483/152324101-255e7edc-f364-4ff5-9e43-8b75f8ff4d00.png) | ![Markup on 2022-02-03 at 11:12:30](https://user-images.githubusercontent.com/25105483/152324156-d3489225-0d6a-4bee-bf23-b9f05b5dd402.png) |
| ![Markup on 2022-02-03 at 11:18:38](https://user-images.githubusercontent.com/25105483/152324202-9609801b-0f11-45e2-90ce-79cf3a0d4692.png) | ![Markup on 2022-02-03 at 11:13:22](https://user-images.githubusercontent.com/25105483/152324241-c9108846-5cd7-4425-856f-d74ae7b404b7.png) |

Screen recording with changes applied:

https://user-images.githubusercontent.com/25105483/152329958-32ab09d9-dacd-4c89-a785-42bacae77b95.mov

#### Changes proposed in this Pull Request:
- add necessary block `supports`: `align`, `color` and `spacing` - in `index.js` and also on the server side (`mailchimp.php`)
- extract block styles and classes from `\WP_Block_Supports::get_instance()->apply_block_supports()`
- block classes are no longer received from `Blocks::classes( FEATURE_NAME, $attr )`

#### Jetpack product discussion
This PR is a part of the _Expanding Jetpack Block design tooling for Full Site Editing_ project: pdDOJh-6-p2.

#### Does this pull request change what data or activity we track or use?
No, it does not.

#### Testing instructions:

The Mailchimp block requires connection to WordPress.com. We can use Jurassic Tube or Jurassic Ninja ([direct link with this branch](https://jurassic.ninja/create?jetpack-beta&branches.jetpack=add/mailchimp-block-align-color-spacing-features&wp-debug-log))

1. Create a new post or a page with the Mailchimp block. It is necessary to have a (test) Mailchimp account registered.
2. There should be a new set of features in the block sidebar: Color and Dimensions / Padding & Margin; plus there should be the Align tool available in the block toolbar
3. Experiment with the new features and observe whether the change applies correctly in the back end
4. Save the changes and review if the settings are rendering correctly in the front end as well
5. We can test the change with multiple themes (block-based, classic). Not all themes support all features. If a theme doesn't support a specific feature, this feature won't be displayed in the block sidebar.

For example, themes like Storefront and Dara don't support block Dimensions / Padding & Margin and therefore it won't display in the block sidebar. Theme like Hever, Maywood or Geologist support both: Color and Dimensions / Padding & Margin.
